src/schedule/ucc_schedule.h

```diff
@@ -142,6 +169,10 @@ static inline ucc_status_t ucc_task_complete(ucc_coll_task_t *task)
         task->cb.cb(task->cb.data, status);
     }
 
+    if (task->executor && (task->flags & UCC_COLL_TASK_FLAG_EXECUTOR_STOP)) {
```

Comment:
do we want to probably stop executor before triggereing user cb? e.g., if user wants to do cuda calls inside cb.

Reply:
agree, changed order